src/report.rs

```diff
@@ -72,8 +75,11 @@ impl Reporter {
                 .write()
                 .expect("failed to obtain write lock for stats");
             let mut snap = StatMap::default();
-            for (k, v) in orig.drain() {
-                snap.insert(k, v);
+            for (k, v) in orig.iter_mut() {
+                let value = ::std::mem::replace(v,
+                                                Histogram::<u64>::new(HISTOGRAM_PRECISION)
+                                                    .expect("failed to build Histogram"));
```

Comment:
Ok, I think finishing this `drain()` implementation is the cleanest way forward.